### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by NeckerFree**
**Project requirements**

- Make sure YAGNI, KISS, and DRY rules and Ruby style guide are respected.
- Follow SOLID principles.
- Pay special attention to the "Single-responsiblity Principle".
- Make sure that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- Extract some pieces of logic into separate class(es).

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -32,25 +32,63 @@ def list_all_people
...
+    name = gets.chomp
+    puts('Enter classroom: ')
+    classroom = gets.chomp
+    puts('Has parent permission? [Y/N]: ')
+    response = gets.chomp
+    permission = response.downcase == 'y'
     student = Student.new(age, classroom, name, parent_permission: permission)
     puts("Created student id: #{student._id} name: #{student._name}")
     @people << student
   end
```
Comment (thecodechaser) [inline app.rb]: - [ ] In `create_student` method you are getting input from users and creating the student, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

File: app.rb [hunk 2]
```diff
@@ -32,25 +32,63 @@ def list_all_people
...
+    puts('Enter age: ')
+    age = gets.chomp
+    puts('Enter specialization: ')
+    specialization = gets.chomp
+    puts('Enter name: ')
+    name = gets.chomp
     teacher = Teacher.new(age, specialization, name)
     @people << teacher
     puts("Created teacher id: #{teacher._id} name: #{teacher._name}")
   end
```
Comment (thecodechaser) [inline app.rb]: - [ ] In `create_teacher` method you are getting input from users and creating the teacher, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

File: app.rb [hunk 3]
```diff
@@ -32,25 +32,63 @@ def list_all_people
...
-  def create_book(title, author)
+  def create_book
+    puts('Enter title: ')
+    title = gets.chomp
+    puts('Enter author: ')
+    author = gets.chomp
     book = Book.new(title, author)
     @books << book
     puts("Created book id: #{book.id} title: #{book.title} author: #{book.author}")
   end
```
Comment (thecodechaser) [inline app.rb]: - [ ] In `create_book` method you are getting input from users and creating the book, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

File: app.rb [hunk 4]
```diff
@@ -32,25 +32,63 @@ def list_all_people
...
+  def create_rental # rubocop:disable  Metrics/MethodLength
+    puts('Enter date (YYYY-MM-DD): ')
+    date = gets.chomp
+    puts('Enter book id: ')
+    book_id = gets.chomp.to_i
+    puts('Enter person id: ')
+    person_id = gets.chomp.to_i
     book = @books.find { |b| b._id == book_id }
     if book.nil?
       puts("book with id: #{book_id} doesn't exist")
```
Comment (thecodechaser) [inline app.rb]: - [ ] In `create_rental` method you are getting input from users and creating the rental, This method does not follow `single responsibility principle`. Please make sure to separate these operations into different methods to achieve `single responsibility principle`.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @NeckerFree, and @amrendrakind👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Whoistolu — APPROVED** [review]
### Hi Team

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**